flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/ZooKeeperCompletedCheckpointStore.java

```diff
@@ -128,7 +145,9 @@ public ZooKeeperCompletedCheckpointStore(
 		// All operations will have the path as root
 		this.client = client.usingNamespace(client.getNamespace() + checkpointsPath);
 
-		this.checkpointsInZooKeeper = new ZooKeeperStateHandleStore<>(this.client, stateStorage);
+		this.checkpointsInZooKeeper = (checkpointsInZooKeeper != null) ?
```

Comment:
I am not super happy with this line. This is why I have given up this way the first time. I'd like to learn whether this style is OK in Flink or we tend to build such a testing constructor in another way.

Reply:
it's a bit icky but non-intrusive enough to be acceptable imo,